src/include/migraphx/op/unique.hpp

```diff
@@ -0,0 +1,340 @@
+    // x_rev_indices: [0, 1, 1, 2, 3, 2] --- x seen in terms of indices of unique sequence y
+    // y_count: [1, 2, 2, 1] -- count at each y_index. sum = len(x)
+
+    // NOTE: y [2, 1, 3, 4]   --- the unique output is constructed from x[y_indices[...]]
+    // Output data structures: y_indices, x_rev_indices, y_count are processed inline.
+
+    template <class T>
+    auto unsorted_uniq_indices(const T& input_data, size_t chunk_sz) const
+    {
+        auto idx_less_fn = [&](auto idx1, auto idx2) {
+            // for efficiency, loop is bypassed for single elements. ct = 1
+            for(size_t ct = chunk_sz; ct > 1; idx1++, idx2++, ct--)
+            {
+                if(input_data[idx1] < input_data[idx2])
+                    return true;
+                else if(input_data[idx1] > input_data[idx2])
+                    return false;
+            }
+            return input_data[idx1] < input_data[idx2];
+        };
+        std::map<size_t, size_t, decltype(idx_less_fn)> uniq_val_map(idx_less_fn);
+
+        std::vector<std::size_t> y_indices;
+        std::vector<std::size_t> x_rev_indices;
+        std::vector<std::size_t> y_count;
+
+        // go through all the elements and add the unique elements into the map..
+        // inline processing for outputs: y_indices, x_rev_indices, y_count
+        size_t count_x = input_data.size();
+        for(size_t f_idx = 0; f_idx < count_x; f_idx += chunk_sz)
+        {
+            auto [itr, added_new] = uniq_val_map.insert({f_idx, y_indices.size()});
+            if(added_new)
+            {
+                y_count.push_back(0);
+                y_indices.push_back(x_rev_indices.size());
+            }
+            y_count[itr->second]++;
+            x_rev_indices.push_back(itr->second);
+        }
+
+        return std::tuple{y_indices, x_rev_indices, y_count};
+    }
+
+    // Default: none. Range: [-rank, rank-1]
+    static constexpr int64_t axis_none = std::numeric_limits<int64_t>::max();
+    int64_t axis                       = axis_none;
+
+    // Default: 1= sorted. 0 = unsorted.
+    int64_t sorted = 1;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.axis, "axis"), f(self.sorted, "sorted"));
+    }
+
+    std::string name() const { return "unique"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1).standard();
+
+        auto& sh_x         = inputs[0];
+        auto lens_x        = sh_x.lens();
+        size_t dim_x       = sh_x.ndim();
+        size_t max_uniq_ct = sh_x.elements();
+
+        int64_t t_axis = axis;
+        if(t_axis != axis_none)
+        {
+            t_axis = migraphx::tune_axis(dim_x, t_axis, name());
+            if(t_axis != 0)
+                MIGRAPHX_THROW("Unique: Only supports axis = 0 or None");
+            max_uniq_ct /= sh_x.strides()[t_axis];
```

Comment:
better to do just pick `max_uniq_ct = sh_x.lens[t_axis];`

Reply:
No, this is not the same thing.. I need strides() here. Not Lens(). Thanks.